packages/vscode-extension/src/exp/index.ts

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import * as vscode from "vscode";
+import { IExperimentationService, IExperimentationTelemetry, TargetPopulation, getExperimentationServiceAsync } from "vscode-tas-client";
+import * as pkg from "./../../package.json";
+import { ExtTelemetry } from "./../telemetry/extTelemetry"
+
+class ExperimentationTelemetry implements IExperimentationTelemetry {
+    private sharedProperties: {[key: string]: string};
+
+    constructor() {
+        this.sharedProperties = {};
+    }
+
+    public setSharedProperty(name: string, value: string): void {
+        this.sharedProperties[name] = value;
+    }
+
+    public postEvent(eventName: string, props: Map<string, string>): void {
```

Comment:
Will Exp set shared properties and postEvent automatically?

Reply:
Yes, these two functions will be called automatically, as this telemetry object is passed to ExperimentationService.